### ISSUE TITLE
Files in git moving to annex, but left uncommited in subsequent save()

### ISSUE THREAD

**Issue by mih** [datalad/datalad#4279]
Here is some example code:

```
(src.pathobj / 'test_mod_file').write_text("Some additional stuff.")
src.save(to_git=True, message="Modified.")

# passes
assert_repo_status(src_repo, annex=annex)

(src.pathobj / 'test_mod_annex_file').write_text("Heavy stuff.")
src.save(to_git=not annex, message="Modified again.")

# fails, reporting modification of `test_mod_file`
assert_repo_status(src_repo, annex=annex)
```

You can see a real-life result here: https://ci.appveyor.com/project/mih/datalad/builds/31380585/job/ggu2uihujmu7na6s#L1131

Of course, this is a probabilistic issue. Moreover, I feel that this has been reported somewhere, and was potentially fixed. But it happens right now with the pinned annex version on our windows boxes 7.20191107-g8ea269ef7.

**Comment by kyleam** [comment]
> Moreover, I feel that this has been reported somewhere, and was potentially fixed. But it happens right now with the pinned annex version on our windows boxes 7.20191107-g8ea269ef7.

This may be the same source of the git->annex conversion that I reported [here][0].  That specific issue was fixed in git-annex 7.20191230.  Our use of the blunt `-c annex.largefiles=anything` override may cause similar git->annex conversion problems, though, but hopefully the switch to `--force-large` in gh-4231 addresses those issues.

[0]: https://git-annex.branchable.com/bugs/A_case_where_file_tracked_by_git_unexpectedly_becomes_annex_pointer_file/

**Comment by mih** [comment]
This has not resurfaced with today's master and a bump of git-annex to 8.20200309 https://github.com/datalad/datalad/pull/4283/files
